jcl/src/java.base/share/classes/com/ibm/tools/attach/target/AttachHandler.java

```diff
@@ -287,15 +290,17 @@ private boolean initialize() throws IOException {
 		if (null == nameProperty) {
 			nameProperty = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties().getProperty("sun.java.command"); //$NON-NLS-1$
 		}
-		
-		if ((null == IPC.logStream) && (null != loggingProperty)
-				&& loggingProperty.equalsIgnoreCase("yes")) { //$NON-NLS-1$
-			File logFile = new File(logName + pidProperty + ".log"); //$NON-NLS-1$
-			logStream = new PrintStream(logFile);
-			IPC.setLogStream(logStream);
-			IPC.loggingEnabled = true;
-			IPC.setDefaultVmId(pidProperty);
-			IPC.logMessage("AttachHandler initialize"); //$NON-NLS-1$
+		synchronized (IPC.accessorMutex) {
+			if ((null == IPC.logStream) && (null != loggingProperty)
+					&& loggingProperty.equalsIgnoreCase("yes")) { //$NON-NLS-1$
+				File logFile = new File(logName + pidProperty + ".log"); //$NON-NLS-1$
+				IPC.logStream = new PrintStream(logFile);
+				IPC.setDefaultVmId(pidProperty);
+				IPC.logMessage("AttachHandler initialize"); //$NON-NLS-1$
```

Comment:
This message won't be logged since logging is not yet enabled.

Reply:
Right.  Changing to unconditional print.

Reply:
Or you can set LOGGING_ENABLED before calling logMessage()